### PR TITLE
`sage.plot`, `sage.repl`: Modularization fixes (imports), `# needs`

### PR DESCRIPTION
**Pull request by mkoeppe**
Also removing a use of `SR`, when testing for `Expression` is good enough.

- Cherry-picked from #35095.

### :memo: Checklist

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: src/sage/plot/hyperbolic_regular_polygon.py [hunk 1]
```diff
@@ -22,9 +22,10 @@
...
 from sage.misc.decorators import options, rename_keyword
-from sage.symbolic.constants import pi, e
-from sage.functions.hyperbolic import arccosh
-from sage.functions.trig import sin, cos, cot
+from sage.misc.lazy_import import lazy_import
+lazy_import("sage.symbolic.constants", ["pi", "e"])
+lazy_import("sage.functions.hyperbolic", "arccosh")
+lazy_import("sage.functions.trig", ["sin", "cos", "cot"])
 from sage.misc.functional import is_odd
 from sage.matrix.constructor import matrix
```
Comment (kwankyu) [inline src/sage/plot/hyperbolic_regular_polygon.py:32]: The same comment before: please separate out `lazy_import` statements from normal imports.

File: src/sage/plot/plot3d/plot3d.py [hunk 2]
```diff
@@ -154,7 +154,8 @@ def f(x,y): return math.exp(x/5)*math.cos(y)
 from sage.plot.colors import rainbow
 from .texture import Texture
 
-from sage.functions.trig import cos, sin
+from sage.misc.lazy_import import lazy_import
+lazy_import("sage.functions.trig", ["cos", "sin"])
 from sage.misc.sageinspect import sage_getargspec, is_function_or_cython_function
```
Comment (kwankyu) [inline src/sage/plot/plot3d/plot3d.py]: the same.

File: src/sage/repl/ipython_kernel/interact.py [hunk 3]
```diff
@@ -173,8 +175,14 @@ def widget_from_single_value(cls, abbrev, *args, **kwds):
 
             return input_grid(abbrev.nrows(), abbrev.ncols(),
                               default=abbrev.list(), to_value=abbrev.parent())
-        if isinstance(abbrev, Color):
-            return SageColorPicker(value=abbrev.html_color())
+
+        try:
+            from sage.plot.colors import Color
+        except ImportError:
```
Comment (kwankyu) [inline src/sage/repl/ipython_kernel/interact.py]: Why import it again? See line 48.

**Comment by kwankyu** [comment]
Otherwise lgtm/

File: src/sage/repl/ipython_kernel/interact.py (hunk 3 above)
Comment (mkoeppe) [reply src/sage/repl/ipython_kernel/interact.py]: Thanks, fixed

File: src/sage/plot/hyperbolic_regular_polygon.py (hunk 1 above)
Comment (mkoeppe) [reply src/sage/plot/hyperbolic_regular_polygon.py:32]: Done.

**Review by kwankyu — APPROVED** [review]
Tests passed.

**Comment by mkoeppe** [comment]
Thanks!

**Pull request merged**